iroh/src/api.rs

```diff
@@ -0,0 +1,93 @@
+use std::collections::{HashMap, HashSet};
+use std::path::Path;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use bytes::Bytes;
+use cid::Cid;
+use iroh_resolver::resolver::Path as IpfsPath;
+use libp2p::gossipsub::MessageId;
+use libp2p::{Multiaddr, PeerId};
+
+pub struct Id {
+    pub peer_id: PeerId,
+    pub listen_addrs: Vec<Multiaddr>,
+    pub local_addrs: Vec<Multiaddr>,
+}
+
+#[derive(Debug)]
+pub enum Ping {
+    PeerId(PeerId),
+    Multiaddr(Multiaddr),
+}
+
+pub trait Api<P: P2p, S: Store>: Accessors<P, S> + GetAdd {}
+
+pub trait Accessors<P: P2p, S: Store> {
+    fn p2p(&self) -> Result<P>;
+    fn store(&self) -> Result<S>;
+}
+
+#[async_trait(?Send)]
```

Comment:
@dignifiedquire here's the `?Send`

Reply:
If we require `GetAdd` to be `Send`, we shouldn't use `?Send` here